test/parse.jl

```diff
@@ -366,8 +366,14 @@ let a⊂b = reduce(&, x ∈ b for x in a) && length(b)>length(a)
     @test !([1,2] ⊂ [1,2])
 end
 
+
 # issue #9503
 @test parse("x<:y") == Expr(:(<:), :x, :y)
 @test parse("x>:y") == Expr(:(>:), :x, :y)
 @test parse("x<:y<:z").head === :comparison
 @test parse("x>:y<:z").head === :comparison
+
+# issue #11169
+@test_throws UndefVarError undefvar
+let undefvar = return end
```

Comment:
This is a bit sketchy --- it's not clear we should even allow `return` without any obvious enclosing function. Also since this is a `let`, it wouldn't lead to `undefvar` being defined globally anyway. Maybe we should test `f(return)` and confirm that `f` wasn't called (via a side effect)?